Sources/PackageCollectionsSigning/Utilities/Utilities.swift

```diff
@@ -0,0 +1,40 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Vapor open source project
+//
+// Copyright (c) 2017-2020 Vapor project authors
+// Licensed under MIT
+//
+// See LICENSE for license information
+//
+// SPDX-License-Identifier: MIT
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+// Source: https://github.com/vapor/jwt-kit/blob/master/Sources/JWTKit/Utilities/Utilities.swift
+
+extension DataProtocol {
+    func copyBytes() -> [UInt8] {
+        if let array = self.withContiguousStorageIfAvailable({ buffer in [UInt8](buffer) }) {
+            return array
+        } else {
+            let buffer = UnsafeMutableBufferPointer<UInt8>.allocate(capacity: self.count)
+            defer { buffer.deallocate() }
+
+            self.copyBytes(to: buffer)
+            return [UInt8](buffer)
```

Comment:
This is more costly than we really need: you can use `[UInt8](self)` directly and that should be fine. Alternatively, if you wanted _maximum speed_ you could do:

```
return Array<UInt8>(unsafeUninitializedCapacity: self.count) { buffer in
    self.copyBytes(to: buffer)
}
```

In principle this could replace the entire implementation of this function as it should be fast regardless of whether the object is contiguous.

Reply:
amended https://github.com/apple/swift-package-manager/pull/3238/commits/6a0489b50be0d4143959ca7e20bbb8d750323b76